File: Modules/SMES/MES/SIE.Web.MES/LoadItems/DeductItems/Commands/WoCostItemAddCommand.cs
using SIE.Domain;
using SIE.MES.LoadItems;
using SIE.Resources.Employees;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SIE.Web.MES.LoadItems.DeductItems.Commands
{
    /// <summary>
    /// 工单耗用单添加命令
    /// </summary>
    public class WoCostItemAddCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        /// <exception cref="NotImplementedException"></exception>
        protected override object Excute(ViewArgs args, string scope)
        {
            var woCostItem = args.Data.ToJsonObject<WoCostItem>();
            var dateTime = RF.Find<WoCostItem>().GetDbTime();
            woCostItem.CostNo = RT.Service.Resolve<WoCostItemController>().GetCostNoRule(1).FirstOrDefault();
            woCostItem.State = SIE.MES.LoadItems.Enum.WoCostItemState.ToSubmit;
            woCostItem.CreateBy = RT.IdentityId;            
            woCostItem.CreateDate = dateTime;
            return woCostItem;
        }
    }
}

File: Modules/SMES/MES/SIE.Web.MES/LoadItems/DeductItems/Commands/WoCostItemAddCommand.js
SIE.defineCommand("SIE.Web.MES.LoadItems.DeductItems.Commands.WoCostItemAddCommand", {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "icon-AddEntity icon-green" },
    onItemCreated: function (entity) {
        var model = entity.model;
        var me = this;
        this.view.execute({
            data: model,
            success: function (res) {
                var data = res.Result;
                if (data.CostNo === null || data.CostNo === "") {
                    var me = this;
                }
                else {
                    entity.setCostNo(data.CostNo);
                    entity.setState(data.State);
                }
            }
        }, me.view);
    },
});